core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -82,44 +100,44 @@ public MatchQueryBuilder fromXContent(QueryParseContext parseContext) throws IOE
                 if (token == XContentParser.Token.FIELD_NAME) {
                     currentFieldName = parser.currentName();
                 } else if (token.isValue()) {
-                    if ("query".equals(currentFieldName)) {
+                    if (parseContext.parseFieldMatcher().match(currentFieldName, QUERY_FIELD)) {
                         value = parser.objectText();
-                    } else if ("type".equals(currentFieldName)) {
+                    } else if (parseContext.parseFieldMatcher().match(currentFieldName, TYPE_FIELD)) {
                         String tStr = parser.text();
-                        if ("boolean".equals(tStr)) {
+                        if (parseContext.parseFieldMatcher().match(tStr, BOOLEAN_FIELD)) {
```

Comment:
While using ParseField also for values is useful, I'm wondering if it might be confusing on the long run. I would at least rename BOOLEAN_FIELD and the following to something else, since technically they are not fields. No strong opinion though.